### DISCUSSION TITLE
Scanner Recommendations for Linux Flatpak Version

### DISCUSSION THREAD

**Discussion by batonac** [cyanfish/naps2#204] [General]
Hi, I really enjoy NAPS2 on Windows, and am super happy to have a Linux version as well. I tried installing the Flatpak version on NixOS, and found that the warning about not supporting 3rd-party drivers applies to me here.

Can anyone point me in the direction of "driverless" scanners that work with Linux? I need to invest in a new desktop scanner, so now's a good time to figure this out.

Thanks

**Comment by cyanfish** [comment]
If you're willing to install the .deb or .rpm version those should work with any Linux scanning drivers (flatpaks are isolated so they only work with the drivers I bundle with it). But the devices that should work in flatpaks are [listed here](http://www.sane-project.org/sane-mfgs.html). Also pretty much any modern network scanner should work as those use the standard ESCL protocol and don't require manufacturer-specific drivers.

**Comment by batonac** [comment]
Hey, thanks for getting back to me so quickly. I wish I could just install the .deb or .rpm versions, but alas, I'm a glutton for punishment, and installed NixOS, which ordinarily has been stellar with software packages, [but yet lacks one for NAPS2](https://github.com/NixOS/nixpkgs/issues/249522). I'll reference the table you linked, thanks!

**Comment by batonac** [comment]
So I acquired an Epson DS-320 with the hope it would work "out of the box" based on the `Complete` designation in the linked SANE supported devices table. Unfortunately, the native scanning apps support the device, but the flatpak version of NAPS2 doesn't. Did I happen to hit on a backend that you decided not to bundle?

**Reply by cyanfish** [reply]
Hmm, that uses the `epsonds` backend which is bundled in the Flatpak and is supposed to work. Are you using it over USB?

**Reply by batonac** [reply]
I think that there may be some kind of conflict between the native SANE service--and/or the avahi - mDNS/DNS-SD daemon--and the flatpak app, because initially the flatpak app picked up another printer/scanner on my network, but the "Choose Device" now fails instantaneously with no results, and this behavior seemed to correspond with installing more native scanning software on my system.

One way or another, I'm hoping to get naps2 packaged for NixOS, I just haven't had time to look into it yet.
